content/blog/vscode-esp-idf-v1.8.0/index.md

```diff
@@ -0,0 +1,114 @@
+
+To use it in a ESP-IDF project `.vscode/launch.json`, you can add this configuration:
+
+```JSON
+{
+  "configurations": [
+    {
+      "type": "gdbtarget",
+      "request": "attach",
+      "name": "Eclipse CDT GDB Adapter"
+    }
+  ]
+}
+```
+
+After you build (**ESP-IDF: Build your Project**), flash (**ESP-IDF: Flash your Project**) you can go to menu `Run` and press `Start debugging`. You can check how to use the new debugger by following the [debug tutorial](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/debugging.md).
+
+{{< figure
+    default=true
+    src="img/debug-example.webp"
+    alt=""
+    caption=""
+    >}}
+
+## Advanced configuration of Eclipse CDT Debug Adapter
+
+There are many configurable options to customize the debug adapter if desired. You can view a deeper description in the [debugging documentation](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/DEBUGGING.md).
+
+## ESP-IDF version switcher
+
+Whenever you use the **ESP-IDF: Configure ESP-IDF Extension** to install ESP-IDF in your system, these installation are saved in our extension as global state IDF setup. You can now use the **ESP-IDF: Select Current ESP-IDF Version** to set in your current project (workspace folder in Visual Studio Code language) the ESP-IDF version you desire to use. If you want to delete these IDF setups from the extension global state you can use the **ESP-IDF: Clear Saved ESP-IDF Setups**.
+
+{{< figure
+    default=true
+    src="img/idf-version-list.webp"
+    alt=""
+    caption=""
+    >}}
+
+The current ESP-IDF version for currently open ESP-IDF project is shown in the status bar items like this:
+
+{{< figure
+    default=true
+    src="img/idf-version-status-bar.webp"
+    alt=""
+    caption=""
+    >}}
+
+## Add ESP-IDF: idf.py reconfigure task command
+
+For code navigation the [Microsoft C/C++ Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) or [Clangd extension](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd) can be used for C/C++ language support. These extensions provide feature such as syntax highlight, go to definition/declaration and other common language feature you would expect in an IDE. Many of these extensions rely on a `compile_commands.json` file, a JSON compilation database which consist of an array of “command objects”, where each command object specifies one way a translation unit is compiled in the project. More information about JSON compilation database can be found [here](https://clang.llvm.org/docs/JSONCompilationDatabase.html).
+
+The **ESP-IDF: idf.py reconfigure task** will execute ESP-IDF reconfigure task, which generate a compile_commands.json so the language extension can work properly. This is useful in case the user doesn't want to fully build the project but wants to see code navigation and other language features.
```

Comment:
Please correct me if I am wrong, but, for now, the first paragraph reads like a long introduction to what code navigation is and how it works followed by a brief and simple introduction to what "ESP-IDF: idf.py reconfigure task" is. For those who know what is `compile_commands.json`, the second paragraph would be anticlamactic, as they would be wondering why they had to waste time on reading the first paragraph if the feature is so simple.

However, I am still not sure why the user might not want to fully build the project? Maybe you can provide good reasons for the readers to consider or to feel related?

I tried to rewrite this section with that idea in mind. Here is what I have got:

> Now you can generate a `compile_commands.json` file for code navigation extensions without fully building your project by using the command `ESP-IDF: idf.py reconfigure task`. Read on for more explanations.
>
> One of the key features of an IDE is code navigation. While working on your project, you usually want to benefit from such common language features as syntax highlight, code completion, go to definition/declaration, and others. For C/C++ language support, usually the extensions like [C/C++ extension][c-extension] or [clangd extension][clangd-extension] are used.
>
> Such extensions usually rely on a `compile_commands.json` file -- a JSON compilation database which consist of an array of *command objects*, where each command object specifies one way in which a translation unit is compiled in the project. More information about a JSON compilation database can be found [here][code-compilation-database].
>
> If, for any reason, you don’t want to fully build your project but still want to use code navigation, you can use the command `ESP-IDF: idf.py reconfigure task` to generate a `compile_commands.json` file for the language extension to work properly. This command executes the ESP-IDF reconfigure task, hence its name.

If you like it, here is the markdown version for it:

[c-extension]: https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools
[clangd-extension]: https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd
[code-compilation-database]: https://clang.llvm.org/docs/JSONCompilationDatabase.html

```markdown
Now you can generate a `compile_commands.json` file for code navigation extensions without fully building your project by using the command `ESP-IDF: idf.py reconfigure task`. Read on for more explanations.

One of the key features of an IDE is code navigation. While working on your project, you usually want to benefit from such common language features as syntax highlight, code completion, go to definition/declaration, and others. For C/C++ language support, usually the extensions like [C/C++ extension][c-extension] or [clangd extension][clangd-extension] are used.

Such extensions usually rely on a `compile_commands.json` file -- a JSON compilation database which consist of an array of *command objects*, where each command object specifies one way in which a translation unit is compiled in the project. More information about a JSON compilation database can be found [here][code-compilation-database].

If, for any reason, you don’t want to fully build your project but still want to use code navigation, you can use the command `ESP-IDF: idf.py reconfigure task` to generate a `compile_commands.json` file for the language extension to work properly. This command executes the ESP-IDF reconfigure task, hence its name.

[c-extension]: https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools
[clangd-extension]: https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd
[code-compilation-database]: https://clang.llvm.org/docs/JSONCompilationDatabase.html
```